Formula/libmarpa.rb

```diff
@@ -0,0 +1,42 @@
+class Libmarpa < Formula
+  desc "Marpa parse engine C library -- STABLE"
+  homepage "https://jeffreykegler.github.io/Marpa-web-site/libmarpa.html"
+  url "https://github.com/jeffreykegler/libmarpa/archive/refs/tags/v8.6.2.tar.gz", using: :nounzip
+  sha256 "b7eb539143959c406ced4a3afdb56419cc5836e679f4094630697e7dd2b7f55a"
+  license "MIT"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "libtool" => :build
+  depends_on "texlive" => :build
+
+  def install
+    ENV.deparallelize
+    system "tar", "-xzf", "libmarpa-" + version.to_s + ".tar.gz"
+    chdir "libmarpa-#{version}" do
+      system "make", "dists"
+    end
+    mkdir "build" do
+      system "cmake", "../libmarpa-" + version.to_s + "/cm_dist", *std_cmake_args, "--install-prefix", prefix.to_s
+      system "cmake", "--build", "."
+      system "cmake", "--install", "."
+      File.rename prefix.to_s + "/inc", include.to_s
+    end
```

Comment:
```suggestion
    system "make", "dists"
    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
    include.install (prefix/"inc").children
```
Doesn't this work without `:nounzip`?

Reply:
That would create an in-tree build, which is not what the library author recommends in his HOWTO document.  I might be able to get that to work… but it might also install some of the build products, since it uses some kind of funky perl script to copy the files.  This formula was pretty delicate to get working at all.

Reply:
Nope, that change doesn't work for me:
```
cp work/stage/libmarpa-`cat LIB_VERSION`.tar.gz .
test -d timestamp || mkdir timestamp
date > timestamp/tar.stamp
sh etc/work_to_dist.sh
cp work/doc/libmarpa-doc-`cat LIB_VERSION`.tar.gz .
sh etc/work_to_doc_dist.sh
cp work/doc1/libmarpa-doc1-`cat LIB_VERSION`.tar.gz .
sh etc/work_to_doc1_dist.sh
perl cmake/to_dist.pl
test -d timestamp || mkdir timestamp
date > timestamp/cm_dist.stamp
==> cmake -S . -B build -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/libmarpa/8.6.2 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_TESTING=OFF -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk
CMake Error: The source directory "/tmp/libmarpa-20220322-50686-1cap2qq/libmarpa-8.6.2" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
/opt/homebrew/Library/Homebrew/shims/shared/git --version
/opt/homebrew/Library/Homebrew/shims/shared/curl --version
/opt/homebrew/Library/Homebrew/ignorable.rb:29:in `block in raise'
BuildError: Failed executing: cmake -S . -B build -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/libmarpa/8.6.2 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_TESTING=OFF -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk
```

Reply:
In fact the CMakeLists.txt contains this:

```
if("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
   message(SEND_ERROR "In-source builds are not allowed.")
   endif("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
```

Reply:
That shouldn't get classified as in-source build since it uses dedicated `build` directory.

There is a slight mistake in suggestion for source directory (should be `cm_dist` not `.`):
```suggestion
    system "make", "dists"
    system "cmake", "-S", "cm_dist", "-B", "build", *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
    include.install (prefix/"inc").children
```

Reply:
That change doesn't work for me either:

```
Last 15 lines from /Users/dave/Library/Logs/Homebrew/libmarpa/01.make:
2022-04-01 23:08:20 +0000

make
dists

make: *** No rule to make target `dists'.  Stop.
```

Did it work for you?

Reply:
Did you remove the `:nounzip` from url?